### PR TITLE
Switch to Scratch Auth

### PR DESCRIPTION
**Pull request by Mbrick2**
FluffyScratch is currently down so this could be useful because @hamptonmoore said he doesn't have time to fix it.
[Here is more info](https://auth.itinerary.eu.org/documentation/migrating-from-fluffyscratch)
**I have not done any testing yet**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -333,9 +333,9 @@ app.post('/api/reactions/:id', cors(corsOptions), async (req, res) => { // react
 
 app.get('/auth/begin', (req, res) => {
     if (req.get('host') == 'localhost:8081') {
-        res.redirect(`https://fluffyscratch.hampton.pw/auth/getKeys/v2?redirect=bG9jYWxob3N0OjgwODEvYXV0aC9oYW5kbGU=`)
+        res.redirect(`https://auth.itinerary.eu.org/auth/?redirect=bG9jYWxob3N0OjgwODEvYXV0aC9oYW5kbGU=&name=Ocular`)
```
Comment (Looky1173) [inline index.js:336]: @Mbrick2 For the localhost base64 encoded URL, you should include the "http" prefix as Scratch Auth assumes "https" when you don't specify a protocol, which wouldn't work on localhost. The correct encoded URL is therefore `aHR0cDovL2xvY2FsaG9zdDo4MDgxL2F1dGgvaGFuZGxl`

**Review by Looky1173 — APPROVED** [review]

**Comment by Looky1173** [comment]
@jeffalo You might want to consider merging this pull request to avoid unnecessary redirects. Also, it seems like FluffyScratch stopped redirecting to Scratch Auth and instead returns a _502 Bad Gateway_ error. You can always revert a merged pull request if you change your mind.

**Comment by jeffalo** [comment]
Thank you. I will take a look when I get some time. :)

File: index.js (hunk 1 above)
Comment (Weredime) [inline index.js:336]:
```suggestion
        res.redirect(`https://auth.itinerary.eu.org/auth/?redirect=bG9jYWxob3N0OjgwODEvYXV0aC9oYW5kbGU=&name=ocular`)
```

**Comment by Looky1173** [comment]
@jeffalo People are complaining on Ocular's forum topic about not being able to sign in. This is because FluffyScratch fails to redirect to Scratch Auth for some reason. Merging this PR would fix this. I understand that you don't have much time to work on Ocular but this PR literally only changes the URL which my-ocular makes auth requests to; there's nothing to review here!

**Comment by jeffalo** [comment]
Thanks. I haven't been able to reproduce any issues with Fluffyscratch, but this should probably avoid them.

**Comment by Looky1173** [comment]
@jeffalo Thank you for merging the PR! Also, have you restarted the my-ocular server yet (as the changes do not seem to be live as of now)?

**Comment by jeffalo** [comment]
Just restarted it now :)

**Comment by jeffalo** [comment]
Reverting because it looks like it's redirecting to localhost for some reason.

File: index.js [hunk 2]
```diff
@@ -333,9 +333,9 @@ app.post('/api/reactions/:id', cors(corsOptions), async (req, res) => { // react
...
     } else {
-        res.redirect(`https://fluffyscratch.hampton.pw/auth/getKeys/v2?redirect=bXktb2N1bGFyLmplZmZhbG8ubmV0L2F1dGgvaGFuZGxl`)
+        res.redirect(`https://auth.itinerary.eu.org/auth/?redirect=aHR0cDovL2xvY2FsaG9zdDo4MDgxL2F1dGgvaGFuZGxl&name=Ocular`)
```
Comment (Looky1173) [inline index.js:338]:
```suggestion
        res.redirect(`https://auth.itinerary.eu.org/auth/?redirect=bXktb2N1bGFyLmplZmZhbG8ubmV0L2F1dGgvaGFuZGxl&name=Ocular`)
```
Fix redirect location

**Comment by Looky1173** [comment]
@jeffalo My [suggested change](https://github.com/jeffalo/my-ocular/pull/31/files#r895222023) fixes the incorrect redirect. I think as the owner of the repo you can accept the suggestion with a click. :-)

**Comment by Looky1173** [comment]
Hi @jeffalo, any updates? :-D My suggested change resolves the issue with the redirect. I can't wait to sign into Ocular once you switch to Scratch Auth!

**Comment by Looky1173** [comment]
Hello @jeffalo, sorry for bothering you with the comments above, I just saw on _wasteof_ that you are on a trip! Enjoy the journey! 🙂

**Comment by Weredime** [comment]
> unsolicited chat link

@Looky1173 Probably you shouldn't reference this as it breaks community guidelines and this repository is linked to on scratch

**Comment by Looky1173** [comment]
@Weredime Well, the name of the site is the same as its domain name... Anyway, I edited my post to not contain the direct URL.

**Comment by Looky1173** [comment]
Hello @jeffalo, I hope you had a pleasant trip. Could you please take a look at my [suggested fix](https://github.com/jeffalo/my-ocular/pull/31#issuecomment-1153267478)?

**Pull request merged**